controller/src/main/java/io/pravega/controller/store/stream/tables/serializers/StateRecordSerializer.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream.tables.serializers;
+
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import io.pravega.controller.store.stream.tables.State;
+import io.pravega.controller.store.stream.tables.StateRecord;
+
+import java.io.IOException;
+
+public class StateRecordSerializer extends VersionedSerializer.WithBuilder<StateRecord, StateRecord.StateRecordBuilder> {
+    @Override
+    protected byte getWriteVersion() {
+        return 0;
+    }
+
+    @Override
+    protected void declareVersions() {
+        version(0).revision(0, this::write00, this::read00);
+    }
+
+    private void read00(RevisionDataInput revisionDataInput, StateRecord.StateRecordBuilder builder) throws IOException {
+        int ordinal = revisionDataInput.readCompactInt();
+        if (ordinal < State.values().length) {
+            builder.state(State.values()[ordinal]);
+        } else {
+            // TODO: what happens if newer states are added and we are trying to deserialize a higher revisioned data from
```

Comment:
Is the current handling introducing a bug in the case we bump up the version?

In any case, need to file an issue for this TODO and reference it here.

Reply:
updated